### PR TITLE
small refactor for plotting experiment results

### PR DESCRIPTION
**Pull request by bjuergens**
* changed plot and batchplot scripts to use TAP.
* Added option for smoothing to batch_generate_plot.py
* some minor changes to existing CLI paramters to both scripts

On a side note: 

I use the following pipeline to look at simulation results

```
cd ~/dev/project/NeuroEvolution-CTRNN_new && python neuro_evolution_ctrnn/batch_generate_plot.py results/data/ --smooth 5 && cd results && python resuts_to_csv.py && libreoffice data.csv
```

This assumes the following: 

* the venv is active
* the repository is located under `~/dev/project/NeuroEvolution-CTRNN_new`
* the repository contains a softlink to the results - repository with the name `results`

### CODE REVIEW

**Comment by bjuergens** [comment]
Can be reviewed and merged now

File: neuro_evolution_ctrnn/batch_generate_plot.py [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
+    filename: str = "plot.png"  # the filename under which to store the plots
+    handle_existing: str = "skip"  # what to do when the plot-file already exists? options: skip, replace, raise
+    smooth: int = 0  # How strong should the lines be smoothed? (0 to disable)
 
-parser.add_argument('--handle-existing', type=str,
-                    help='what to do when the plot-file already exists? options: skip, replace, raise',
-                    default='skip')
+    def configure(self):
+        self.description = "iterate over simulation folders and create a plot for each simulation run by" \
+                           "by calling plot_experiment.py for each folder"
```
Comment (pdeubel) [inline neuro_evolution_ctrnn/batch_generate_plot.py:20]:
```suggestion
        self.description = """iterate over simulation folders and create a plot for each simulation run by
                           calling plot_experiment.py for each folder"""
```

File: neuro_evolution_ctrnn/batch_generate_plot.py [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
+        self.add_argument("basedir")
 
-simulation_folders = [f.path for f in os.scandir(args.base_dir) if f.is_dir()]
+        # aliases
+        self.add_argument("-e", "--handle_existing")
+
+
+args = BatchPlotArgs(underscores_to_dashes=True).parse_args()
+
+simulation_folders = [f.path for f in os.scandir(args.basedir) if f.is_dir()]
```
Comment (pdeubel) [inline neuro_evolution_ctrnn/batch_generate_plot.py:32]:
```suggestion
simulation_folders = [f.path for f in os.scandir(args.basedir)]
```

Das if kann weg denke ich -> kriege so oder so ein error wenn args.basedir kein Folder ist

**Review by pdeubel — APPROVED** [review]
Das eine if kann mMn weg ansonsten passts 👍🏽

File: neuro_evolution_ctrnn/batch_generate_plot.py (hunk 2 above)
Comment (bjuergens) [reply neuro_evolution_ctrnn/batch_generate_plot.py:32]: ich denke das ist da, für so einen fall

/base_dir/simulationresult1/
/base_dir/simulationresult2/
/base_dir/simulationresult2/
/base_dir/file.txt

Da soll die file.txt dann keinen fehler werfen

**Pull request merged**